group_project_v2/mixins.py

```diff
@@ -232,10 +246,58 @@ def studio_view(self, context):  # pylint: disable=unused-argument, no-self-use
         return fragment
 
 
-class DashboardMixin(object):
+class DashboardXBlockMixin(object):
+    """ Mixin for an XBlock that has dashboard views """
     def dashboard_view(self, context):
         raise NotImplementedError(MUST_BE_OVERRIDDEN)
 
+    def dashboard_detail_view(self, context):
+        raise NotImplementedError(MUST_BE_OVERRIDDEN)
```

Comment:
@bradenmacdonald `dashboard_details_view` is not used anywhere yet - it will be used to provide detailed view of group status (similar to Group Work Status Report): see [docs (Group Work detail)|https://mckinseyacademy.atlassian.net/wiki/display/PROD/McKA+Admin+Group+Work+Status+and+Dashboard].

So, this follows the same approach as other views, i.e. navigation view. To be more precise, `dashboard_view` is what would be shown in (docs)main view, dashboard_details_view is for (docs)details view.

The idea with that `raise NotImplementedError(MUST_BE_OVERRIDDEN)` is to indicate that this class is abstract one (I've tried to use abc, it didn't work for the reason I can't recall right now - likely something to do with XBlock using its own metaclasses). So, the class that would not have this view would just return nothing, or raise another exception.

Actually, both project, activity and stage have dashboard_view; dashboard_details_view is available for activity and stage, but not project. So there'll be just one "exception".